airflow/providers/apache/iceberg/provider.yaml

```diff
@@ -31,7 +31,7 @@ dependencies:
   - apache-airflow>=2.7.0
 
 devel-dependencies:
-  - pyiceberg>=0.5.0
+  - pyiceberg>=0.5.1
```

Comment:
It might be good to require a later version, for example, PyIceberg 0.6.1. The project is improving quickly, and each release has a ton of bugfixes.